README.md

```diff
@@ -1,95 +1,66 @@
 
-# Activate the virtual environment and make sure that 'wheel' is installed
-source .venv/bin/activate
-pip install --upgrade wheel pip setuptools
+### System dependencies
 
-# Install all the ICON4Py packages and its dependencies
-# External dependencies would be checked out at './_external_src'
-pip install --src _external_src -r requirements-dev.txt
+ICON4Py requires **_Python >= 3.10_** and **_boost >= 1.85.0_**, and uses the `uv` tool to manage the development workflow. `uv` is a versatile tool which bundles together functionality from different applications: it can work as a _fast_ Python package manager (like `pip`), as a dependency version exporter (like `pip-tools`), as a Python application runner (like `pipx`) or as a full project development manager (like `hatch`). 
+`uv` can be installed in different ways (check its [installation instructions](https://docs.astral.sh/uv/getting-started/installation/)), like using the recommended standalone installer:
 
-# Finally, check that everything works
-pytest -v
+```bash
+$ curl -LsSf https://astral.sh/uv/install.sh | sh 
 ```
+although it can also be installed with `pip` / `pipx`
 
-The `--src _external_src` option tells `pip` to use a specific folder as the base path for checked out sources, which is very convenient for development tasks involving changes in external dependencies like `gt4py`. For convenience, `./_external_src` has been already added to the repository `.gitignore`.
+```bash
+$ pipx install uv
+```
 
-You can also use [tox](https://tox.wiki/en/latest/) for the automatic installation of all packages in development mode in a single step:
+### ICON4Py development environment
 
+Once `uv` is installed in your system, it is enough to clone this repository and let `uv` handling the installation of the development environment.
 
 ```bash
 # Clone the repository
 git clone git@github.com:C2SM/icon4py.git
 cd icon4py
 
-# Use tox to create and set up a development environment (usually at `.venv`) in verbose mode
-pip install tox
-python -m tox -vv -e dev --devenv .venv
+# Let uv create the development environment at `.venv`.
+# The `--extra all` option tells uv to install all optional
+# dependencies of icon4py so it is not strictly necessary
+uv sync --extra all
 
-# Activate the virtual environment and check that everything works
+# Activate the virtual environment and start writing code!
 source .venv/bin/activate
-pytest -v
 ```
 
+The new `venv` is a standard Python virtual environment with all the necessary run-time and development dependencies, where all the icon4py subpackages have been already installed in editable mode. New packages can be installed with the `uv pip` subcommand which emulates the `pip` interface or even using the regular `pip` which is also available in the venv (although it's usually much slower and not really recommended).   
```

Comment:
```suggestion
The new `venv` is a standard Python virtual environment with all the necessary run-time and development dependencies, where pre-installed icon4py subpackages in editable mode. New packages can be installed with the `uv pip` subcommand which emulates the `pip` interface or even using the regular `pip` which is also available in the venv (although it's usually much slower and not really recommended).   
```

Reply:
I slightly changed the whole paragraph and split it into two. Check the new version.